services/ui-src/src/components/fields/DateField.tsx

```diff
@@ -14,16 +14,20 @@ export const DateField = ({ name, label, sxOverride, ...props }: Props) => {
   const form = useFormContext();
   form.register(name);
 
-  const [displayValue, setDisplayValue] = useState("");
+  const [displayValue, setDisplayValue] = useState<string>("");
+  const [formattedValue, setFormattedValue] = useState<string>("");
 
-  useEffect(() => {
-    if (displayValue) {
-      form.setValue(name, displayValue || "", { shouldValidate: true });
+  const onChangeHandler = (rawValue: string, formattedValue: string) => {
+    setDisplayValue(rawValue);
+    setFormattedValue(formattedValue);
+    const completeDate = checkDateCompleteness(formattedValue);
+    if (completeDate) {
+      form.setValue(name, formattedValue, { shouldValidate: true });
     }
-  }, [displayValue]);
+  };
```

Comment:
setting both values in local state, then checking if the date is complete. only if it is complete will it run a validation check. this is to stop the previous behavior of yelling at a user immediately when they start entering a date, for instance just typing `9`. now it will wait to set the form value and validate until it is either 1) a complete date or 2) they leave the field prior to entering a complete date, triggering `onBlur`